packages/next/client/index.tsx

```diff
@@ -271,7 +278,8 @@ export async function initNext(opts: { webpackHMR?: any } = {}) {
 
     const { component: app, exports: mod } = appEntrypoint
     CachedApp = app as AppComponent
-    if (mod && mod.reportWebVitals) {
+    const exportedReportWebVitals = mod && mod.reportWebVitals
+    if (exportedReportWebVitals) {
```

Comment:
@devknoll it's here, when we try to define the `onPerfEntry`, we only know the exports of app module but not the used hooks since it's not executed yet, I didn't think of a good condition to check for hooks usage. If we changed the condition to `true`, the app will collect perf anyway, I guess we don't want that?

Reply:
The thing is, I don't think this saves us much. We still bundle `web-vitals` and it still executes, it just throws away the values that it computed. It seems like it's probably fine to just always compute and buffer them, and in the worst case scenario dump that buffer later (e.g. after a navigation or timeout) if there are no listeners registered.

Reply:
Another thing worth mentioning is that I _think_ both `web-vitals` and our use of `onPerfEntry` just use buffered performance data anyway, so maybe the whole thing can be lazy when any callback is added...